### PR TITLE
UI for supporting ABAC in the dashboard

### PR DESCRIPTION
**Pull request by joshenlim**
- Contains ABAC for org + projects
- ABAC billing + members will come in a subsequent PR
  - Members role management is locked to reflect the current behaviour (new members will be defaulted to the "Developer" role id when invited to organization)
- Permission checking is behind a feature flag `enablePermissions` (all permissions will pass through if flag is off)
- Removed usage of permissions checking in places which are not within the scope (anything outside of orgs, projects, billing or members)

### CODE REVIEW

File: studio/hooks/queries/usePermissions.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  let organization_id: number | undefined
+  const { ref, slug } = router.query
+  if (ref) {
+    const project = find(app.projects.list(), { ref }) as Project | undefined
+    organization_id = project?.organization_id
+  } else if (slug) {
+    const organization = find(app.projects.list(), { ref }) as Organization | undefined
+    organization_id = organization?.id
+  }
```
Comment (alaister) [inline studio/hooks/queries/usePermissions.ts:46]: Doesn't look like `organization_id` is used? I'm assuming it's here for future use?

File: studio/components/interfaces/Organization/NewProject/NotOrganizationOwnerWarning.tsx [hunk 2]
```diff
@@ -8,12 +8,11 @@ const NotOrganizationOwnerWarning = () => {
         icon={<IconAlertCircle className="text-white" size="large" strokeWidth={1.5} />}
         defaultVisibility={true}
         hideCollapse
-        title="You do not have permission to create a project"
+        title="You do not have permissions to create a project"
```
Comment (alaister) [inline studio/components/interfaces/Organization/NewProject/NotOrganizationOwnerWarning.tsx:11]: nit: I think the old one is maybe more correct?
Or maybe: You do not have _the correct_ permissions to create a project

**Review by alaister — COMMENTED** [review]
Looks good!

File: studio/hooks/queries/usePermissions.ts (hunk 1 above)
Comment (joshenlim) [reply studio/hooks/queries/usePermissions.ts]: whoa sharp eyes!

File: studio/components/interfaces/Organization/NewProject/NotOrganizationOwnerWarning.tsx (hunk 2 above)
Comment (joshenlim) [reply studio/components/interfaces/Organization/NewProject/NotOrganizationOwnerWarning.tsx]: yeap! thats fair, will go with the former

**Review by alaister — APPROVED** [review]
lgtm!

**Pull request merged**